python/src/HSICEstimatorImplementation_doc.i.in

```diff
@@ -0,0 +1,284 @@
+%define OT_HSICEstimator_doc
+"Base class of HSICEstimator.
+
+Available constructors:
+    HSICEstimator()
+
+    HSICEstimator(covarianceList, X, Y, estimatorType)
+
+    HSICEstimator(covarianceList, X, Y, estimatorType, weightFunction)
```

Comment:
no need to list constructors here as they have the same arguments, just mention that weightFunction is optional